examples/DivisorCurve.jl

```diff
@@ -0,0 +1,329 @@
+"""
+function multiplicity(C::AffinePlaneCurve{S}, F::Oscar.MPolyElem{S}, P::Point{S}) where S <: FieldElem
+    f = divrem(F, C.eq)
+    if iszero(f[2])
+        return Inf
+    elseif isconstant(f[2])
+        return 0
+    else
+        return intersection_multiplicity(C, AffinePlaneCurve(f[2]), P)
+    end
+end
+
+################################################################################
+
+@doc Markdown.doc"""
+    multiplicity(C::ProjPlaneCurve{S}, F::Oscar.MPolyElem_dec{S}, P::Oscar.Geometry.ProjSpcElem{S}) where S <: FieldElem
+
+Return the multiplicity of the polynomial `F` on the curve `C` at the point `P`.
+"""
+function multiplicity(C::ProjPlaneCurve{S}, F::Oscar.MPolyElem_dec{S}, P::Oscar.Geometry.ProjSpcElem{S}) where S <: FieldElem
+    f = divrem(F.f, defining_equation(C))
+    if iszero(f[2])
+        return Inf
+    elseif isconstant(f[2])
+        return 0
+    else
+        R = parent(C.eq)
+        return intersection_multiplicity(C, ProjPlaneCurve(R(f[2])), P)
+    end
+end
+
+################################################################################
+################################################################################
+
+@doc Markdown.doc"""
+    divisor(C::AffinePlaneCurve{S}, F::Oscar.MPolyElem{S}) where S <: FieldElem
+
+Return the divisor defined by the polynomial `F` on the curve `C`.
+"""
+function divisor(C::AffinePlaneCurve{S}, F::Oscar.MPolyElem{S}) where S <: FieldElem
+    f = divrem(F, C.eq)
+    !iszero(f[2]) || error("The polynomial is zero on the curve")
+    E = Dict{Point{S}, Int}()
+    if !isconstant(f[2])
+        CC = AffinePlaneCurve(f[2])
+        L = curve_intersect(C, CC)
+        for P in L[2]
+            E[P] = intersection_multiplicity(C, CC, P)
+        end
+    end
+    return AffineCurveDivisor(C, E)
+end
+
+################################################################################
+
+@doc Markdown.doc"""
+    divisor(C::AffinePlaneCurve{S}, phi::AbstractAlgebra.Generic.Frac) where S <: FieldElem
+
+Return the divisor defined by the rational function `phi` on the curve `C`.
+"""
+function divisor(C::AffinePlaneCurve{S}, phi::AbstractAlgebra.Generic.Frac) where S <: FieldElem
+    g = divrem(phi.den, C.eq)
+    !iszero(g[2]) || error("This is not a rational function on the curve")
+    f = divrem(phi.num, C.eq)
+    !iszero(f[2]) || error("The numerator is zero on the curve")
+    E = Dict{Point{S}, Int}()
+    if !isconstant(f[2])
+        CN = AffinePlaneCurve(f[2])
+        LN = curve_intersect(C, CN)
+        for P in LN[2]
+            E[P] = intersection_multiplicity(C, CN, P)
+        end
+    end
+    F = Dict{Point{S}, Int}()
+    if !isconstant(g[2])
+        CD = AffinePlaneCurve(g[2])
+        LD = curve_intersect(C, CD)
+        for P in LD[2]
+            F[P] = -intersection_multiplicity(C, CD, P)
+        end
+    end
+    return AffineCurveDivisor(C, mergewith(+, E, F))
```

Comment:
Again, I might be missing something, but couldn't this be simplified to the following:
```suggestion
    return divisor(C, f) - divisor(C, g)
```